interpreter/code_interpreters/languages/shell.py

```diff
@@ -54,4 +54,15 @@ def add_active_line_prints(code):
     for index, line in enumerate(lines):
         # Insert the echo command before the actual line
         lines[index] = f'echo "## active_line {index + 1} ##"\n{line}'
-    return '\n'.join(lines)
\ No newline at end of file
+    return '\n'.join(lines)
+
+
+def wrap_in_trap(code):
```

Comment:
I think we removed this in c93eb678c6bd87cd94f4ed86f9ba11d3d22a3ded as it doesn't properly emit an "end_of_execution" flag (`exit` here will fire before it reaches that) so if it errors it just hangs forever.